packages/runtime/src/useCases/common/RuntimeErrors.ts

```diff
@@ -276,6 +276,19 @@ class IdentityDeletionProcess {
     }
 }
 
+class IdentityMetadata {
+    public recordNotFound() {
+        return new ApplicationError("error.runtime.identityMetadata.recordNotFound", "There is no stored IdentityMetadata for the specified combination of reference and key.");
+    }
+
+    public unknownReferencedIdentity() {
+        return new ApplicationError(
+            "error.runtime.identityMetadata.unknownReferencedIdentity",
+            "The reference of the IdentityMetadata does not resolve to the address of a known Identity. IdentityMetadata can only be upserted for the own Identity or for the peers of the Relationships."
```

Comment:
Nothing about the code `unknownReferencedIdentity` makes clear that this can only be used for the upsert usecase. Maybe we can either make the code clearer or make the message a bit more generic.

=> imagine we check for an existing identity in get / delete and throw an error there, you would just pick this error based on the method name